department-of-veterans-affairs/vets-website
# Authed patterns - orange task add review page

## Are you removing, renaming or moving a folder in this PR?

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

- Main change is to add the new review page to the 22-1990 form aka task Orange
  - This design is a major departure from the existing accordion and should be treated as an 'alpha' prototype
  - Will be creating several tickets for revisions, but the main functionality for the task requirements are present
- Added a page for updating the 'other contact info' items to form schema
- Added 'review' properties to form schema to allow dynamic rendering of review sections on the new review page
- updated useMockedLogin so that it doesn't fire in production or staging allowing staging app to work with staging users as expected (once they are logged in with normal login flow)
- Edit navigation is added for 'edit as subtask' section of this task. This is needed so that the save button can reroute back to review page


## Related issue(s)

- https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/159

## Testing done

- Tested locally
- Steps
  - navigate to http://localhost:3001/mock-form-ae-design-patterns/2/task-orange/introduction?loggedIn=true
  - you should be able to add a service history period to that section if desired. Granted the UI doesn't look quite like the design yet since we haven't determined its final state, yet so the existing list and loop pattern is used
  - verify that app can be clicked through, and that review page is shown with form information
  - click edit link for 'Other contact information' and verify that it takes you to an edit page for that information
  - make an update to email and verify the email address and press save and verify that email is changed on the review page and that the 'other contact information' section of the review page is focused upon returning (still need to trigger appropriate validation for required fields on the edit page fyi, so it is still possible to not re-enter email when updating)

## Remaining items / existing issues to resolve (these will be broken into their own tickets)
I made a mini epic for the things that need to be updated from newest design and functionality related. https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/170


## Screenshots
![orange-other-contact-info](https://github.com/user-attachments/assets/d8a4a358-f1e1-4677-81d1-a2bd61141a03)

![orange-other-contact-info-edit](https://github.com/user-attachments/assets/3641aa1e-3085-4124-8369-1d4a3e060271)


![orange-personal-info](https://github.com/user-attachments/assets/293e1671-410a-4911-892b-97fe60c78e15)


![orange-review-updated](https://github.com/user-attachments/assets/c22a82f3-0551-4754-b6ae-d56ca7050528)

## What areas of the site does it impact?

AE design pattern mock form, not production facing

## Acceptance criteria

Review page is present
Updating other contact info from review page redirects to edit page and updates form data

### Quality Assurance & Testing


### Error Handling


### Authentication
